examples/c/ksyscall.bpf.c

```diff
@@ -0,0 +1,27 @@
+#include "vmlinux.h"
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_tracing.h>
+#include <bpf/bpf_core_read.h>
+
+#define TASK_COMM_LEN 16
+
+SEC("ksyscall/kill")
+int BPF_KSYSCALL(entry_probe, pid_t tpid, int sig) 
+{
+
+	if (sig == 0) {
+		/**
+			If sig is 0, then no signal is sent, but existence and permission
+			checks are still performed; this can be used to check for the
+			existence of a process ID or process group ID that the caller is
+			permitted to signal.
+		*/
+		return 0;
+	}
+
+	bpf_get_current_comm(&comm, sizeof(comm));
+	bpf_printk("KILL syscall called on %d (%s) with signal: %d.", tpid, comm, sig);
```

Comment:
so I get that KILL was requested for `tpid`, but it wasn't really requested by `comm`, right? It's quite misleading. comm is from current process that is issuing kill() syscall. So let's make it a bit clearer. And also log pid of the calling process, just for the fun of it?

Something like: "KILL syscall called by PID %d (%s) for PID %d"